url9512d8fa89_bdkosher_card_games_tgz-pJ8-org_washcom_cardgames_battleroyale_BattleRoyaleGameSimulatorJ8/src/org/washcom/cardgames/battleroyale/ThreeBattlerAssessor.java

```diff
@@ -0,0 +1,120 @@
+package org.washcom.cardgames.battleroyale;
+
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import static org.washcom.cardgames.battleroyale.Rules.*;
+import static org.washcom.cardgames.core.Denomination.ACE;
+import static org.washcom.cardgames.core.Denomination.TWO;
+
+/**
+ * A BattleAssessor for when there's exactly three battlers in the battle. If
+ * not, the methods will throw IllegalArgumentExceptions.
+ *
+ * @author Joe
+ */
+public class ThreeBattlerAssessor implements BattleAssessor {
+
+    private static final Logger log = Logger.getLogger(ThreeBattlerAssessor.class.toString());
+
+    @Override
+    public BattleCard pickWinner(Battle battle) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 62-76 into a separate method.